### PR TITLE
Introduce RUBY_PATCHNAME constant

### PR DESCRIPTION
**Pull request by shayonj**
`RUBY_PATCHNAME` will return:

- p{level}
- dev
- rc/preview labels

This is especially helpful when ruby is in
rc/preview and `RUBY_PATCHLEVEL` is set to `-1`.

`RUBY_PATCHNAME` is only a proposal name for the constant :)
Feature request at https://bugs.ruby-lang.org/issues/14217

### CODE REVIEW

File: version.c [hunk 1]
```diff
@@ -59,6 +60,12 @@ Init_version(void)
      * the patchlevel will be -1
      */
     rb_define_global_const("RUBY_PATCHLEVEL", MKINT(patchlevel));
+    /*
+     * The patchlevel string for this ruby. If this is a development build
+     * of ruby the patchlevel string will be dev otherwise the respective
```
Comment (r00ster91) [inline version.c]:
```suggestion
     * of ruby, the patchlevel string will be dev, otherwise the respective
```

**Comment by ioquatix** [comment]
This seems like a nice idea.. but what about just having a 4th version number, e.g. `RUBY_VERSION = 2.5.2.157`

**Comment by k0kubun** [comment]
It seems to have a conflict now. Could you rebase this from master?

**Comment by k0kubun** [comment]
Let me close this as it has not been updated for a while. Please reopen this after resolving conflicts. Thanks.

**Pull request closed without merging**